src/agent.ts

```diff
@@ -1,92 +1,51 @@
 import {
-  BlockEvent,
   Finding,
-  Initialize,
-  HandleBlock,
-  HealthCheck,
   HandleTransaction,
-  HandleAlert,
-  AlertEvent,
   TransactionEvent,
-  FindingSeverity,
-  FindingType,
+  FindingSeverity, 
+  FindingType,    
 } from "forta-agent";
+import { FORTA_BOT_REGISTRY, BOT_DEPLOYER_ADDRESS, NEW_AGENT_FUNCTION_SIGNATURE } from "./constants";
+
+
+export default function provideHandleTransaction(
+  proxyAddress: string,
+  deployerAddress: string,
+  functionABI: string
+): HandleTransaction {
+  return async (txEvent: TransactionEvent):Promise<Finding[]>  => {
+    const findings: Finding[] = [];
+    //check from address with deployer address
+    if (txEvent.from != deployerAddress.toLowerCase()) {
+      return findings;
+    }
 
-export const ERC20_TRANSFER_EVENT =
-  "event Transfer(address indexed from, address indexed to, uint256 value)";
-export const TETHER_ADDRESS = "0xdAC17F958D2ee523a2206206994597C13D831ec7";
-export const TETHER_DECIMALS = 6;
-let findingsCount = 0;
-
-const handleTransaction: HandleTransaction = async (
-  txEvent: TransactionEvent
-) => {
-  const findings: Finding[] = [];
-
-  // limiting this agent to emit only 5 findings so that the alert feed is not spammed
-  if (findingsCount >= 5) return findings;
-
-  // filter the transaction logs for Tether transfer events
-  const tetherTransferEvents = txEvent.filterLog(
-    ERC20_TRANSFER_EVENT,
-    TETHER_ADDRESS
-  );
-
-  tetherTransferEvents.forEach((transferEvent) => {
-    // extract transfer event arguments
-    const { to, from, value } = transferEvent.args;
-    // shift decimals of transfer value
-    const normalizedValue = value.div(10 ** TETHER_DECIMALS);
+    // Store/filter bot transactions
+    const botTxs = txEvent.filterFunction(functionABI, proxyAddress);
 
-    // if more than 10,000 Tether were transferred, report it
-    if (normalizedValue.gt(10000)) {
+    // Iterate through transactions
+    botTxs.forEach((tx) => {
+      const { agentId, owner, chainIds, metadata } = tx.args;
+      
+      //Create a Finding object and push it into the findings array
       findings.push(
         Finding.fromObject({
-          name: "High Tether Transfer",
-          description: `High amount of USDT transferred: ${normalizedValue}`,
-          alertId: "FORTA-1",
-          severity: FindingSeverity.Low,
-          type: FindingType.Info,
+          name: "Nethermind Bot Deployment Detector",
+          description: "Detects Bots Deployed by Nethermind",
+          alertId: "FORTA-123",
+          severity: FindingSeverity.Info, 
+          type: FindingType.Info,        
           metadata: {
-            to,
-            from,
+            agentId:agentId.toString(),  
+            owner,
+            chainIds:chainIds.toString(),
+            metadata: metadata,
```

Comment:
```suggestion
            metadata,
```
Just a small suggestion. You already did it on L40 with `owner`, so can do the shorthand here as well.